doc/kernel/usermode/syscalls.rst

```diff
@@ -189,6 +189,26 @@ generated due to its repetitive nature and can be found in
 ``include/generated/syscall_macros.h``. It is created by
 ``scripts/gen_syscall_header.py``.
 
+The final layer is the invocation of the system call itself. All architectures
+implementing system calls must implement the seven inline functions
+:c:func:`_arch_syscall_invoke0` through :c:func:`_arch_syscall_invoke6`.  These
+functions marshal arguments into designated CPU registers and perform the
+necessary privilege elevation. In this layer all arguments are treated as an
+unsigned 32-bit type, and there is always a 32-bit unsigned return value, which
```

Comment:
End the sentence and start a new one:   "unsigned 32-bit type.  There is always..."